packages/flutter/lib/src/material/date.dart

```diff
@@ -8,9 +8,282 @@
 library;
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
 
 import 'material_localizations.dart';
 
+/// A delegate that supplies localized strings and other values for the
+/// date picker.
+abstract class DatePickerDelegate {
+  /// Creates a date picker delegate.
+  const DatePickerDelegate();
+
+  /// Returns a [DateTime] representing the current date and time.
+  DateTime now();
+
+  /// Returns a [DateTime] with the date of the original, but time set to
+  /// midnight.
+  DateTime dateOnly(covariant DateTime date);
+
+  /// Returns a [DateTimeRange] with the dates of the original, but with times
+  /// set to midnight.
+  ///
+  /// See also:
+  ///  * [dateOnly], which does the same thing for a single date.
+  DateTimeRange datesOnly(covariant DateTimeRange range);
+
+  /// Returns true if the two [DateTime] objects have the same day, month, and
+  /// year, or are both null.InheritedWidget
+  bool isSameDay(covariant DateTime? dateA, covariant DateTime? dateB);
+
+  /// Returns true if the two [DateTime] objects have the same month and
+  /// year, or are both null.
+  bool isSameMonth(covariant DateTime? dateA, covariant DateTime? dateB);
+
+  /// Determines the number of months between two [DateTime] objects.
+  ///
+  /// For example:
+  ///
+  /// ```dart
+  /// DateTime date1 = DateTime(2019, 6, 15);
+  /// DateTime date2 = DateTime(2020, 1, 15);
+  /// int delta = DateUtils.monthDelta(date1, date2);
+  /// ```
+  ///
+  /// The value for `delta` would be `7`.
+  int monthDelta(covariant DateTime startDate, covariant DateTime endDate);
+
+  /// Returns a [DateTime] that is [monthDate] with the added number
+  /// of months and the day set to 1 and time set to midnight.
+  ///
+  /// For example:
+  ///
+  /// ```dart
+  /// DateTime date = DateTime(2019, 1, 15);
+  /// DateTime futureDate = DateUtils.addMonthsToMonthDate(date, 3);
+  /// ```
+  ///
+  /// `date` would be January 15, 2019.
+  /// `futureDate` would be April 1, 2019 since it adds 3 months.
+  DateTime addMonthsToMonthDate(covariant DateTime monthDate, int monthsToAdd);
```

Comment:
Can this handle calendars with more than 12 months if it uses an extension on DateTime for its dates?

Reply:
Yes, because DateTime doesn't enforce a fixed month count.